### PR TITLE
hashing trick+ RP filters

### PR DESCRIPTION
**Pull request by marouabahri**
Adding the hashing trick and random projection filters.

### CODE REVIEW

File: moa/src/main/java/moa/streams/filters/HashingTrickFilter.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
+
+
+package moa.streams.filters;
+import com.github.javacliparser.IntOption;
+import com.google.common.hash.Hashing;
```
Comment (garawalid) [inline moa/src/main/java/moa/streams/filters/HashingTrickFilter.java:5]: You forget to include the guava dependency.
At `moa/moa/pom.xml` add the following dependency:
```xml
    <dependency>
      <groupId>com.google.guava</groupId>
      <artifactId>guava</artifactId>
      <version>29.0-jre</version>
    </dependency>
```

File: moa/src/main/java/moa/streams/filters/HashingTrickFilter.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    }
+
+    @Override
+    public void getDescription(StringBuilder sb, int indent) {
+        // TODO Auto-generated method stub
+    }
+
+
+    public  double[] hashVector(Instance instance, int n,
+                                com.google.common.hash.HashFunction hashFunction) {
```
Comment (garawalid) [inline moa/src/main/java/moa/streams/filters/HashingTrickFilter.java:91]: I'd write: 
```java
import com.google.common.hash.HashFunction;
/*
.
.
*/

 public  double[] hashVector(Instance instance, int n, HashFunction hashFunction) {
```

**Review by garawalid — CHANGES_REQUESTED** [review]
@marouabahri 
I left you some comments, otherwise, It looks good to me.  
Also, It would be nicer if we have some unit tests for this class.

**Review by garawalid — APPROVED** [review]
LGTM

File: moa/src/main/java/moa/streams/filters/HashingTrickFilter.java (hunk 2 above)
Comment (marouabahri) [reply moa/src/main/java/moa/streams/filters/HashingTrickFilter.java]: Done, many thanks Walid :)

**Pull request merged**